src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/BuildIntegratedProjectSystem.cs

```diff
@@ -285,5 +296,69 @@ private static IEnumerable<Reference> GetProjectReferences(EnvDTEProject project
                 }
             }
         }
+
+        /// <summary>
+        /// Get the unique names of all references which have ReferenceOutputAssembly set to false.
+        /// </summary>
+        private static async Task<List<string>> GetExcludedReferences(
+            EnvDTEProject project,
+            IVsEnumHierarchyItemsFactory itemsFactory)
+        {
+            var excludedReferences = new List<string>();
+
+            var hierarchy = VsHierarchyUtility.ToVsHierarchy(project);
+            var buildPropertyStorage = hierarchy as IVsBuildPropertyStorage;
+
+            // Get all items in the hierarchy, this includes project references, files, and everything else.
+            IEnumHierarchyItems items;
+            if (ErrorHandler.Succeeded(itemsFactory.EnumHierarchyItems(
+                hierarchy,
+                (uint)__VSEHI.VSEHI_Leaf,
+                (uint)VSConstants.VSITEMID.Root,
+                out items)))
+            {
+                // Loop through all items
+                uint fetched;
+                VSITEMSELECTION[] item = new VSITEMSELECTION[1];
+                while (ErrorHandler.Succeeded(items.Next(1, item, out fetched)) && fetched == 1)
+                {
+                    // Check if the item has ReferenceOutputAssembly
+                    string value;
+                    if (ErrorHandler.Succeeded(buildPropertyStorage.GetItemAttribute(
+                            item[0].itemid,
+                            "ReferenceOutputAssembly",
+                            out value))
+                        && value != null)
+                    {
+                        // We only need to go farther if the flag exists and is set to false
+                        if (string.Equals(value, "false", StringComparison.OrdinalIgnoreCase) ||
+                            string.Equals(value, "off", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
While false, off, and 0 might all be falsey type values for MSBuild, we rarely (if ever) rely on that aspect of expression evaluation, since if it's the empty string an error would result in the build. Instead, MSBuild expressions tend to be `!= 'true'` or `== 'false'` style. Either it's true, or false, or not true or not false. In particular, a quick check of the MSBuild .targets reveal that for this metadata, all the assignments and checks are against the value of `true`. It's set to true if unset, and all checks are whether it is `true`. So the best thing here is probably to replace all three of these tests with: `!string.Equals(value, "true", StringComparison.OrdinalIgnoreCase)`